lib/widgets/login.dart

```diff
@@ -305,46 +363,31 @@ class _PasswordLoginPageState extends State<PasswordLoginPage> {
       }
 
       // TODO(server-7): Rely on user_id from fetchApiKey.
-      final int userId = result.userId ?? await _getUserId(result);
+      final int userId = result.userId ?? await widget.loginPageState._getUserId(result.email, result.apiKey);
       // https://github.com/dart-lang/linter/issues/4007
       // ignore: use_build_context_synchronously
       if (!context.mounted) {
         return;
       }
 
-      final globalStore = GlobalStoreWidget.of(context);
-      // TODO(#108): give feedback to user on SQL exception, like dupe realm+user
-      final accountId = await globalStore.insertAccount(AccountsCompanion.insert(
-        realmUrl: realmUrl,
+      widget.loginPageState._tryInsertAccountAndNavigate(
```

Comment:
```suggestion
      await widget.loginPageState._tryInsertAccountAndNavigate(
```

Otherwise the `finally` block happens sooner than it had been.